src/main/java/com/study/gamereviewrest/controller/GameController.java

```diff
@@ -0,0 +1,50 @@
+package com.study.gamereviewrest.controller;
+
+import com.study.gamereviewrest.exceptionHangler.GameNotFoundException;
+import com.study.gamereviewrest.model.Game;
+import com.study.gamereviewrest.model.Review;
+import com.study.gamereviewrest.repository.GameRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class GameController {
+    @Autowired
+    private GameRepository gameRepository;
+    @GetMapping("/review")
+    public List<Game> getGame(){
+        return gameRepository.findAll();
+    }
+    @PostMapping("/review")
+    public Game createGameReview(@RequestBody Game game){
+        return gameRepository.save(game);
+    }
+
+    @PutMapping("/review/{gameName}")
+    public ResponseEntity<Game> addGameReview(@PathVariable (value = "gameName") String gameName, @RequestBody Review review) throws GameNotFoundException{           //, @RequestBody Review review
+        Game game = gameRepository.findByGameName(gameName);
+        if(game == null){
+            throw new GameNotFoundException("Game does not exist for giving the review!! Create new entry for Game" + gameName);
```

Comment:
you shouldn't be doing a null check in the controller, it is the repository's responsibility
the `findByGameName` method should throw this exception